src/components/check-your-phone/check-your-phone-controller.ts

```diff
@@ -115,6 +72,104 @@ export function checkYourPhonePost(
   };
 }
 
+async function handleMfaChange(
+  intent: Intent,
+  newPhoneNumber: string,
+  updateInput: UpdateInformationInput,
+  sessionDetails: any,
+  req: Request,
+  service: CheckYourPhoneServiceInterface
+): Promise<boolean> {
+  if (intent === INTENT_CHANGE_PHONE_NUMBER) {
+    return handleChangePhoneNumber(
+      newPhoneNumber,
+      updateInput,
+      sessionDetails,
+      req,
+      service
+    );
+  } else if (intent === INTENT_ADD_MFA_METHOD) {
+    return handleAddMfaMethod(
+      newPhoneNumber,
+      updateInput,
+      sessionDetails,
+      req,
+      service
+    );
+  } else {
+    // TODO: Confirm with UCD which error we should display
+    throw Error(`Unknown phone verification intent ${intent}`);
+  }
+}
+
+async function handleChangePhoneNumber(
+  newPhoneNumber: string,
+  updateInput: UpdateInformationInput,
+  sessionDetails: any,
+  req: Request,
+  service: CheckYourPhoneServiceInterface
+): Promise<boolean> {
+  const smsMFAMethod = req.session.mfaMethods.find(
+    (mfa) => mfa.method.mfaMethodType === "SMS"
+  );
+  if (smsMFAMethod) {
+    smsMFAMethod.method.endPoint = newPhoneNumber;
+    updateInput.mfaMethod = smsMFAMethod;
+    return await service.updatePhoneNumberWithMfaApi(
+      updateInput,
+      sessionDetails
+    );
+  } else {
+    // TODO: Confirm with UCD which error we should display
+    throw Error(`No existing MFA method for: ${req.session.user.email}`);
```

Comment:
We can't include the email address in the error message since it'll end up in the logs. We have access to [the trace ID](https://github.com/govuk-one-login/di-account-management-frontend/blob/main/src/middleware/set-local-vars-middleware.ts#L41) at this point - can you switch to that?